### PR TITLE
[11.0][FIX]hotel:Fixed room category issue.

### LINKED ISSUE #51: [11.0] invalid form view sequence in folio line room no and also different category
Steps :
Create New folio 
In  Room Lines click on **Add and item** 
Select Room No And open its view

hotel-issue-03

hotel-issue-01

hotel-issue-02

### PR DESCRIPTION
**Pull request by dhruvitaSerpentCS**
fixed #51

### CODE REVIEW

File: hotel/models/hotel_room_type.py [hunk 1]
```diff
@@ -8,11 +8,31 @@ class HotelRoomType(models.Model):
...
     child_ids = fields.One2many(
         "hotel.room.type", "categ_id", "Child Categories"
     )
+    product_categ_id = fields.Many2one(
+        "product.category", "Product Category", delegate=True
+    )
+
+    @api.model
+    def create(self, vals):
+        if "categ_id" in vals:
```
Comment (JayVora-SerpentCS) [inline hotel/models/hotel_room_type.py]: should be if vals.get('categ_id', False):

File: hotel/models/hotel_room_type.py [hunk 2]
```diff
@@ -8,11 +8,31 @@ class HotelRoomType(models.Model):
...
+            room_categ = self.env["hotel.room.type"].browse(
+                vals.get("categ_id")
+            )
+            vals.update({"parent_id": room_categ.product_categ_id.id})
+        return super(HotelRoomType, self).create(vals)
+
+    @api.multi
+    def write(self, vals):
+        if "categ_id" in vals:
```
Comment (JayVora-SerpentCS) [inline hotel/models/hotel_room_type.py]: same if here

**Review by JayVora-SerpentCS — CHANGES_REQUESTED** [review]

File: hotel/models/hotel_room_type.py (hunk 1 above)
Comment (dhruvitaSerpentCS) [reply hotel/models/hotel_room_type.py]: Done.

File: hotel/models/hotel_room_type.py (hunk 2 above)
Comment (dhruvitaSerpentCS) [reply hotel/models/hotel_room_type.py]: Done.

**Comment by robinkeunen** [comment]
edit (sorry, that was a problem in the coverage and not in tests)

File: hotel/models/hotel_room_type.py (hunk 1 above)
Comment (robinkeunen) [reply hotel/models/hotel_room_type.py]: Why do you prefer `vals.get('categ_id', False)`? I feel that `if "categ_id" in vals` is more readable and _pythonesque_.

**Pull request merged**